### ISSUE TITLE
Add debugging step points for "when" clauses in pattern matching

### ISSUE THREAD

**Issue by dsyme** [dotnet/fsharp#11979]
At the moment we don't lay down any debug points for "when" clauses in pattern matching, so you can't break on them unless there is some sequential logic inside (e.g. `let`, sequencing, loops etc.) 

There is no reason for this - it is simple enough to allow debug points here.

e.g.
```fsharp
let test1 x y =
    match x with
    | [_] when y > 4 -> 5
    | [_] when y < 4 -> -5
    | _ -> 2
```

Check can you validate a breakpoint on the `when` (before debug launch), and can you hit it at runtime.

**Comment by dsyme** [comment]
This is implemented in #11981 though there is an apparent glitch with union patterns
```fsharp
    let TestMatchWithWhen x  =
        match x with
        | [_;x] 
        | [x] when x < 4 -> -5  // The debugpoint for the 'when' is not hit for a two-element list
        | _ -> 2


    TestMatchWithWhen [4;5]
    TestMatchWithWhen [5;4]
    TestMatchWithWhen [6]
```

**Comment by dsyme** [comment]
Fixed in #11981
